gem/tests/test_auth.py

```diff
@@ -441,3 +447,291 @@ def test_auth_middleware(self):
     #     self.assertContains(response, "Welcome to the GEM Wagtail CMS")
     #     # test that the correct logout button is in in the template
     #     self.assertContains(response, "oidc/logout")
+
+
+class TestAllAuth(GemTestCaseMixin, TestCase):
+    def setUp(self):
+        self.user = get_user_model().objects.create_superuser(
+            username='superuser', email='superuser@email.com', password='pass')
+        self.main = self.mk_main(
+            title='main1', slug='main1',
+            path='00010002', url_path='/main1/'
+        )
+        self.site = self.main.get_site()
+
+    @override_settings(ENABLE_ALL_AUTH=True)
+    def test_admin_login_view(self):
+        res = self.client.get(reverse('wagtailadmin_login'))
+        self.assertEqual(res.status_code, 200)
+        self.assertEqual(settings.ENABLE_ALL_AUTH, True)
+
+    def test_admin_login_view_default_allauth_setting(self):
+        res = self.client.get(reverse('wagtailadmin_login'))
+        self.assertEqual(res.status_code, 200)
+        self.assertEqual(settings.ENABLE_ALL_AUTH, False)
+
+    @override_settings(ENABLE_ALL_AUTH=True)
+    def test_admin_views_authed_user(self):
+        self.client.force_login(self.user)
+        res = self.client.get(reverse('wagtailadmin_login'))
+        self.assertEqual(res.status_code, 302)
+        self.assertEqual(settings.ENABLE_ALL_AUTH, True)
+        self.assertEqual(res.url, '/admin/')
+
+        res = self.client.get(res.url)
+        self.assertEqual(res.status_code, 200)
+
+    @override_settings(ENABLE_ALL_AUTH=False)
+    def test_login_all_auth_disabled(self):
+        res = self.client.get(reverse('wagtailadmin_login'))
+        self.assertEqual(res.status_code, 200)
+        self.assertEqual(settings.ENABLE_ALL_AUTH, False)
```

Comment:
This test is pretty much exactly the same as `test_admin_login_view_default_allauth_setting`
Did you mean to log the user in first?